cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -416,11 +419,23 @@ private Metadata getSnapshotBeforeTime(Id.NamespacedId targetId, long timeMillis
    *
    * @param namespaceId the namespace to search in
    * @param searchQuery the search query, which could be of two forms: [key]:[value] or just [value] and can have '*'
-   * at the end for a prefixed search.
-   * @param type the {@link MetadataSearchTargetType} to restrict the search to
+   *                    at the end for a prefix search
+   * @param type the {@link MetadataSearchTargetType} to restrict the search to. For 
+   *             {@link MetadataSearchTargetType#ALL} and {@link MetadataSearchTargetType#ARTIFACT}, returns
+   *             matching artifacts in both {@link ArtifactScope#SYSTEM} and {@link ArtifactScope#USER}
    */
-  public List<MetadataEntry> search(final String namespaceId, final String searchQuery,
-                                    final MetadataSearchTargetType type) {
+  public List<MetadataEntry> search(String namespaceId, String searchQuery, MetadataSearchTargetType type) {
+    List<MetadataEntry> searchResults = new ArrayList<>();
+    searchResults.addAll(doSearch(namespaceId, searchQuery, type));
+    // for MetadataSearchTargetType.ARTIFACT and MetadataSearchTargetType.ALL, also search system scope artifacts
+    if (MetadataSearchTargetType.ARTIFACT.equals(type) || MetadataSearchTargetType.ALL.equals(type)) {
```

Comment:
oh this has nothing to do with ``SYSTEM`` and ``USER`` Metadata Scope. This is for ``ArtifactScope``. The issue is that when you are in default namespace for example, and you do a ``list artifacts``, you see both ``ArtifactScope.SYSTEM`` and ``ArtifactScope.USER`` artifacts. However, when you search metadata, without this PR, only ``ArtifactScope.USER`` artifacts are surfaced in the search results.